src/main/java/io/personium/common/auth/token/AbstractLocalToken.java

```diff
@@ -192,29 +206,29 @@ final String doCreateTokenString(final String[] extendedFields) {
     }
 
     /**
-     * 指定のIssuer向けのIV (Initial Vector)を生成して返します.
-     * IVとしてissuerの最後の最後の１６文字を逆転させた文字列を用います。
-     * これにより、違うIssuerを想定してパースすると、パースに失敗する。
+     * Generate an IV (Initial Vector) and return it for a specified token issuer.
      * @param issuer Issuer URL
      * @return Initial Vector Byte array
      */
     protected static byte[] getIvBytes(final String issuer) {
         try {
-            return StringUtils.reverse("123456789abcdefg" + issuer)
-                    .substring(0, IV_BYTE_LENGTH).getBytes(CharEncoding.UTF_8);
+            MessageDigest md = MessageDigest.getInstance(SHA256);
+            byte[] hash = md.digest(issuer.getBytes(CharEncoding.UTF_8));
+            return Arrays.copyOfRange(hash, IV_OFFSET_FROM_SHA256, IV_OFFSET_FROM_SHA256 + IV_BYTE_LENGTH);
```

Comment:
Why is IV_OFFSET_FROM_SHA256 needed?  I think simply  `Arrays.copyOfRange(hash, 0, IV_BYTE_LENGTH);` is OK.

Reply:
Simply Arrays.copyOfRange(hash, 0, IV_BYTE_LENGTH) will work, of course. 
I think this offset will be for extra bit of security.  

If attacker somehow;  
1. knows that IV is created with SHA256
2. but does not know about the offset.
   (Actually both are obvious by seeing this source, but 1 can also be achieved simply by guessing since SHA256 is too popular. )

Then this offset may confuse the attacker who tries to guess the AES key.